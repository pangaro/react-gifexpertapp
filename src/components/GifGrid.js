import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category } ) => {

    const { data:images, loading } = useFetchGifs( category );

    //console.log(state);
/*
    const [images, setimages] = useState([]);

    useEffect( () => {
        getGifs( category )
            .then( setimages );
            //.then( ( imgs ) => setimages( imgs ) );
    }, [ category ]);
*/

    return (
        <>
            <h3 className="animate__animated animate__swing__delay-2s"> { category } </h3>

            { loading && <p>Loading...</p> }
            <div className="card-grid">
                {
                     images.map( ( img ) => (
                    <GifGridItem 
                        key={ img.id }
                        { ...img } /> 
                    ))
                }
            </div>
        </>
    )
}
