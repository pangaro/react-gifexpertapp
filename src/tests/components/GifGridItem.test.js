import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';


describe('pruebas en <GifGridItem />', () => {

    const title = 'algos';

    const url = 'https://localhost/algo.jpg';
    
    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );


    test('debe mostrar <GifGridItem /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe tener un <P> con el titulo', () => {
        
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
    });
    
    test('debe tener la imagen igual url y alt de los props', () => {
        
        const img = wrapper.find('img');
        //console.log(img.html());

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });
    
    test('el div debe tener la clase animate__swing', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__swing') ).toBe(true);
    });
})
