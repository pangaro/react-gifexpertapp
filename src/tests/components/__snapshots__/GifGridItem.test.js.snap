// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__swing"
>
  <img
    alt="algos"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    algos
     
  </p>
</div>
`;
